include/dr/detail/ranges.hpp

```diff
@@ -152,8 +152,10 @@ struct local_fn_ {
              std::contiguous_iterator<rng::iterator_t<R>> ||
              rng::contiguous_range<R>)
   auto operator()(R &&r) const {
-    if constexpr (has_local_method<rng::iterator_t<R>>) {
-      return std::span(rng::begin(r).local(), rng::size(r));
+    if constexpr (has_local_method<R>) {
+      return r.local();
+    } else if constexpr (has_local_method<rng::iterator_t<R>>) {
+      return rng::views::counted(rng::begin(r).local(), rng::size(r));
     } else if constexpr (has_local_adl<R>) {
       return local_(std::forward<R>(r));
     } else if constexpr (std::contiguous_iterator<rng::iterator_t<R>>) {
```

Comment:
If we prefer `rng::views::counted` over `std::span` let's change also this second place.

I understand motivation here is to not use views from std, but from ranges library, as views from std are not compatible with views from ranges-v3.

But I think `subrange(begin,end)` may be better - it should not need to iterate twice over `forward_range` to first get its size and then find its end (see code [here](https://github.com/ericniebler/range-v3/blob/master/include/range/v3/view/counted.hpp#L104)).

Reply:
counted is a subrange for random access iterators and it also handles non random access.

Reply:
Can [subrange(Iterator && i, Sentinel s)](https://github.com/ericniebler/range-v3/blob/master/include/range/v3/view/subrange.hpp#L211) handle non-random access? Looks like it can. I've not checked it. If so, then it is more efficient.